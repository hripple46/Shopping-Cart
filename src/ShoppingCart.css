.navbar {
  display: flex;
  justify-content: space-between;
  list-style: none;
  padding: 0px;
}

a {
  text-decoration: none;
}
.navbar > li:hover {
  cursor: pointer;
}
.container {
  display: flex;
}
.inventory {
  flex: 2;
}
.cart {
  flex: 1;
}
.inventory > h1 {
  margin-top: 22px;
}
.cart > h2 {
  margin-top: 22px;
}
.itemCounter {
  display: flex;
  justify-content: space-around;
  max-width: 50%;
}
