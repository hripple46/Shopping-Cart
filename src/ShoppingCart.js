import React, { useState } from "react";
import { Link } from "react-router-dom";

let ShoppingCart = () => {
  let [items, setItems] = useState([
    { name: "Banana", qty: 0 },
    { name: "Apple", qty: 0 },
  ]);
  let displayInventory = () => {
    return items.map((item) => {
      return (
        <div key={item.name}>
          <h1>{item.name}</h1>
          <h3>{item.qty}</h3>
        </div>
      );
    });
  };
  return (
    <div>
      <h1>Welcome to Your Shopping Cart</h1>
      {displayInventory()}
      <Link to="/">Home Page</Link>
    </div>
  );
};

export default ShoppingCart;
