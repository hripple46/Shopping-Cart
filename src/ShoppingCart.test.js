import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import ShoppingCart from "./ShoppingCart";
import { act } from "react-dom/test-utils";

describe("ShoppingCart component", () => {
  it("renders correct qty in cart", () => {
    const { container } = render(
      <BrowserRouter>
        <ShoppingCart />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders updated qty", () => {
    render(
      <BrowserRouter>
        <ShoppingCart />
      </BrowserRouter>
    );
    let button = screen.getByRole("button", { name: "add Banana to cart" });
    act(() => {
      userEvent.click(button);
    });

    expect(screen.getByRole("heading", { level: 4 }).textContent).toMatch(
      /banana,2/i
    );
  });
});
