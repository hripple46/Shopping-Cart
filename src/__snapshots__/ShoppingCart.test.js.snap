// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCart component renders correct qty in cart 1`] = `
<div>
  <div>
    <nav>
      <ul
        class="navbar"
      >
        <li>
          <a
            href="/"
          >
            Home Page
          </a>
        </li>
        <li>
          Shopping Cart
        </li>
      </ul>
    </nav>
    <div
      class="container"
    >
      <div
        class="inventory"
      >
        <h1>
          Welcome to Your Shopping Cart
        </h1>
        <div>
          <h1>
            Banana
          </h1>
          <div
            class="itemCounter"
          >
            <button
              id="BananaDec"
            >
              -
            </button>
            <h3>
              2
            </h3>
            <button
              id="Banana"
            >
              +
            </button>
            <button
              id="BananaAddToCart"
            >
              add 
              Banana
               to cart
            </button>
          </div>
        </div>
        <div>
          <h1>
            Apple
          </h1>
          <div
            class="itemCounter"
          >
            <button
              id="AppleDec"
            >
              -
            </button>
            <h3>
              0
            </h3>
            <button
              id="Apple"
            >
              +
            </button>
            <button
              id="AppleAddToCart"
            >
              add 
              Apple
               to cart
            </button>
          </div>
        </div>
      </div>
      <div
        class="cart"
      >
        <h2>
          Cart Will Go Here
        </h2>
      </div>
    </div>
  </div>
</div>
`;
